Chapter_1_Introduction.md

```diff
@@ -0,0 +1,424 @@
+It is an **imperative** style that can be compared to procedural programming.
+It is structured, clear, and optimized for scenarios where the steps are more technical
+or detailed and where the amount of keywords called within a test or tasks are more.
+Also is this style better for complex tasks or complex data
+due to a clear separation between the keyword names and its argument values.
+
+- **Behavior-Driven Style** emphasizes **how the system behaves** from the user's point of view,
+using more natural language and focusing on expected outcomes.
+It is a **declarative** style that can be compared to writing user stories or acceptance criteria.
+It is optimized for **business-oriented** descriptions of functionality
+and is often more suitable for communicating with non-technical stakeholders.
+This style can get less understandable when the amount of steps increases
+or the amount of defined data in the steps increases.
+
+Both styles can be applied within Robot Framework, offering flexibility depending on the context of the automation task.
+
+
+
+### 1.4.4 Data-Driven Specification
+
+> [!IMPORTANT]
+> LO-1.4.4 Recall the purpose of Data-Driven Specification (K1)
+
+**Data-Driven Specification** originates from **Data-Driven Testing**
+and is a method where the test data and expected results are
+separated from the test script that controls the flow.
+
+While in **Robotic Process Automation (RPA)**, the data
+used in an automation workflow is typically acquired dynamically from an external source,
+in testing, the data is specifically chosen to cover different scenarios or cases.
+Therefore, this method of defining data combinations
+statically in the suite files is normally not applicable to RPA.
+
+The purpose of **Data-Driven Testing** is to automate the same sequence of actions
+or scenario with different sets of input and/or expected output data.
+
+In this style, a single user keyword, which contains the whole test logic or sequence of actions,
+is executed with multiple data variations,
+making it highly effective for repetitive tests,
+where the logic stays the same but the data changes,
+without duplicating the test logic for each case.
+
+Robot Framework offers a convenient feature for this approach through **Test Templates**.
+
+**Benefits of Data-Driven Specification**:
+- **Efficiency**: Reduces the need to write redundant test cases by reusing the same workflow with different data inputs.
+- **Clarity**: Keeps the test logic separate from the data, making it easier to manage large data sets.
+- **Scalability**: Suitable for scenarios where the same functionality needs to be tested under various conditions, such as verifying form inputs or performing calculations with different values.
+
+See [3.4 Data-Driven Specification](Chapter_3_Keyword_Design_Variables_Resources.md#34-data-driven-specification) for more details and examples on Data-Driven Specification.
```

Comment:
This renders as a link to this same heading (Chrome Version 131.0.6778.265, MacOS Sonoma 14.7.2)
![image](https://github.com/user-attachments/assets/36ee5e2d-a2de-438e-935f-498a0adc8a22)